File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/category/repository/CategoryRepository.java
package com.side.tiggle.domain.category.repository;

import com.side.tiggle.domain.category.model.Category;
import com.side.tiggle.domain.category.model.CategoryType;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface CategoryRepository extends JpaRepository<Category, Long> {
    List<Category> findByType(CategoryType type);
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/category/repository/CategoryRepository.kt
package com.side.tiggle.domain.category.repository

import com.side.tiggle.domain.category.model.Category
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface CategoryRepository: JpaRepository<Category, Long> {
    fun findCategoryByMemberIdOrDefaults(memberId: Long, default: Boolean): List<Category>
}
